docs/index.rst

```diff
@@ -3,17 +3,95 @@
 
 The ``pytest`` plugin system
 ****************************
+
 ``pluggy`` is the crystallized core of `plugin management and hook
-calling`_ for `pytest`_.
+calling`_ for `pytest`_. It gives users the possibility to extend or modify
+the behaviour of a host program (e.g. `pytest`_, `tox`_ or `devpi`_) by
+installing a `plugin` for that program. The plugin code will run as part of
+normal program execution changing or enhancing certain aspects of it.
 
 In fact, ``pytest`` is itself composed as a set of ``pluggy`` plugins
 which are invoked in sequence according to a well defined set of protocols.
 Some `200+ plugins`_ use ``pluggy`` to extend and customize ``pytest``'s default behaviour.
 
 In essence, ``pluggy`` enables function `hooking`_ so you can build "pluggable" systems.
 
-How's it work?
---------------
+Introduction
+------------
+
+To explain how pluggy can be put to use we need to introduce three roles and explain their interaction with pluggy:
```

Comment:
`To explain how ``pluggy`` is best utilized we need to introduce three use cases`:

A suggestion: reading this over again I think it might be more clear to talk about the code/programs themselves instead of people who write or use the code. This way we talk about what `host` versus `implementer` (maybe just `plugin`?) versus `user` (or `client`) code should look like and how it fits into the `pluggy` use-case model. 

Also, too many references to `pluggy` in a few places when we already know what's being referenced.

Reply:
Good suggestions, I agree. 👍